File: src/main/java/it/giaquinto/springberry/SpringBerryApplication.java
package it.giaquinto.springberry;

import it.giaquinto.springberry.component.SpringBerryLoggerComponent;
import it.giaquinto.springberry.model.builder.connector.SpringBerryConnectorBuilder;
import it.giaquinto.springberry.model.log.LogMessageFactory;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

import java.util.concurrent.Executor;

@SpringBootApplication
@EnableAsync
public class SpringBerryApplication{

    public static void main(String[] args) {
        SpringApplication.run(SpringBerryApplication.class, args);
        SpringBerryLoggerComponent.instance().startLog();
    }

    @Bean
    public TomcatServletWebServerFactory servletContainer() {
        final TomcatServletWebServerFactory tomcatServletWebServerFactory = new TomcatServletWebServerFactory();
        tomcatServletWebServerFactory.addAdditionalTomcatConnectors(new SpringBerryConnectorBuilder().defaultBuild());
        return tomcatServletWebServerFactory;
    }

}

File: src/main/java/it/giaquinto/springberry/SpringBerryApplication.kt
package it.giaquinto.springberry

import it.giaquinto.springberry.raspberry.controller.RaspberryGPIOManager
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication

@SpringBootApplication
open class SpringBerryApplication {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(SpringBerryApplication::class.java, *args)
        }
    }
}